src/core/src/service_interfaces/TelemetryWriter.py

```diff
@@ -75,14 +77,14 @@ def write_package_info(self, package_name, package_ver, package_size, install_du
     # Composed payload
     def write_machine_config_info(self):
         # Machine info - sent only once at the start of the run
-        machine_info = {
+        TelemetryWriter.machine_info = {
             'platform_name': str(self.env_layer.platform.linux_distribution()[0]),
             'platform_version': str(self.env_layer.platform.linux_distribution()[1]),
             'machine_cpu': self.get_machine_processor(),
             'machine_arch': str(self.env_layer.platform.machine()),
             'disk_type': self.get_disk_type()
         }
```

Comment:
rename this function to accurately represent what it is now going to be doing - **set_and_log_machine_config_info**

Reply:
fixed